core/templates/rid_owner.h

```diff
@@ -173,7 +174,12 @@ class RID_Alloc : public RID_AllocBase {
 
 		uint32_t validator = uint32_t(id >> 32);
 
-		if (unlikely(p_initialize)) {
+		if (unlikely(validator >= 0x7FFFFFFF)) {
+			if (THREAD_SAFE) {
```

Comment:
Again, I suspect this check is redundant because if the `validator_chunks` is checked before being set, it shouldn't be `0x80000000` or above?